test/problems/poisson_regression.jl

```diff
@@ -0,0 +1,239 @@
+
+        #Generate Precompute
+        precomp = OptimizationMethods.PrecomputePoissReg(progData)
+
+        #Check Field Type and Dimensions 
+        @test typeof(precomp.obs_obs_t) == Array{real_type, 3}
+        @test size(precomp.obs_obs_t) == (nobs_default, nvar_default, nvar_default)
+        #Compare values 
+        @test reduce(+, precomp.obs_obs_t, dims=1)[1,:,:] ≈ 
+            progData.design' * progData.design atol=
+            eps(real_type) * nobs_default * nvar_default
+
+    end
+
+    ####################################
+    # Test Struct: Allocation
+    ####################################
+
+    # Check if struct is defined 
+    @test isdefined(OptimizationMethods, :AllocatePoissReg)
+
+    # Test super type 
+    @test supertype(OptimizationMethods.AllocatePoissReg) == 
+        OptimizationMethods.AbstractProblemAllocate
+
+    # Test Fields 
+    for name in [:linear_effect, :predicted_rates, :residuals, :grad, :hess]
+        @test name in fieldnames(OptimizationMethods.AllocatePoissReg)
+    end
+
+    # Test Constructors 
+    real_types = [Float16, Float32, Float64]
+    nobs_default = 1000 
+    nvar_default = 50 
+
+    for real_type in real_types 
+        #Generate Random Problem 
+        progData = OptimizationMethods.PoissonRegression(real_type)
+
+        #Generate store 
+        store = OptimizationMethods.AllocatePoissReg(progData)
+
+        #Check Field Type and Dimension: linear_effect
+        @test typeof(store.linear_effect) == Vector{real_type}
+        @test length(store.linear_effect) == nobs_default
+
+        #Check Field Type and Dimension: predicted_rates
+        @test typeof(store.predicted_rates) == Vector{real_type}
+        @test length(store.predicted_rates) == nobs_default
+
+        #Check Field Type and Dimension: residuals
+        @test typeof(store.residuals) == Vector{real_type}
+        @test length(store.residuals) == nobs_default
+
+        #Check Field Type and Dimension: grad
+        @test typeof(store.grad) == Vector{real_type}
+        @test length(store.grad) == nvar_default
+
+        #Check Field Type and Dimension: hess
+        @test typeof(store.hess) == Matrix{real_type}
+        @test size(store.hess) == (nvar_default, nvar_default)
+    end
+
+    ####################################
+    # Test Method: Initialize
+    ####################################
+    real_types = [Float16, Float32, Float64]
+    nobs_default = 1000 
+    nvar_default = 50 
+
+    for real_type in real_types 
+        progData = OptimizationMethods.PoissonRegression(real_type)
+        precomp, store = OptimizationMethods.initialize(progData)
+
+        @test typeof(precomp) == OptimizationMethods.PrecomputePoissReg{real_type}
+        @test typeof(store) == OptimizationMethods.AllocatePoissReg{real_type}
+    end
+
+    ####################################
+    # Test Methods 
+    ####################################
```

Comment:
This section should also include tests to make sure `counters` is being incremented properly

Reply:
Included test for checking counters